src/main/scala/com/gyoho/adget/utils/HttpClient.scala

```diff
@@ -0,0 +1,42 @@
+package com.gyoho.adget.utils
+
+import java.net.URL
+
+import com.twitter.finagle.{Http, Service, http}
+import com.twitter.util.{Future => TwitterFuture}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future, Promise, ExecutionContext}
+
+// use URL instead of String b/c I don't want to deal with edge cases handling
+trait HttpClient[T] {
+  def httpGet(url: URL, timeoutInMillis: Long)(implicit ec: ExecutionContext): T
+  def httpPost(url: URL, payload: String, timeoutInMillis: Long)(implicit ec: ExecutionContext): T
+}
+
+object ScalaHttpClient extends HttpClient[String] {
+  @throws(classOf[java.io.IOException])
+  override def httpGet(url: URL, timeoutInMillis: Long)(implicit ec: ExecutionContext) : String = {
+    val responseFuture = Future(scala.io.Source.fromURL(url).mkString)
+    Await.result(responseFuture, timeoutInMillis.millisecond)
```

Comment:
`scala.concurrent.Await` is a blocking call. Blocking on a future is not recommended because it wastes a thread. Consider to use `after` method from Akka.